lib/serve/rest-api/src/api/endpoints/v2/litellm_passthrough.py

```diff
@@ -106,3 +143,37 @@ async def litellm_passthrough(request: Request, api_path: str) -> Response:
         return StreamingResponse(generate_response(response.iter_lines()))
     else:  # not a streaming request
         return JSONResponse(requests.request(method=http_method, url=litellm_path, json=params, headers=headers).json())
+
+
+def refresh_management_tokens() -> list[str]:
+    """Return DDB entry for token if it exists."""
+    secret_tokens = []
+
+    try:
+        secret_tokens.append(
+            secrets_manager.get_secret_value(SecretId=os.environ.get("MANAGEMENT_KEY_NAME"), VersionStage="AWSCURRENT")[
+                "SecretString"
+            ]
+        )
+        secret_tokens.append(
+            secrets_manager.get_secret_value(
+                SecretId=os.environ.get("MANAGEMENT_KEY_NAME"), VersionStage="AWSPREVIOUS"
+            )["SecretString"]
+        )
+    except Exception:
+        logger.info(f"No previous secret version for {os.environ.get('MANAGEMENT_KEY_NAME')}")
+
+    return secret_tokens
+
+
+def is_valid_management_token(headers: dict[str, str]) -> bool:
+    """Return if API Token from request headers is valid if found."""
+    secret_tokens = refresh_management_tokens()
+    token = get_authorization_token(headers=headers, header_name="Authorization").strip()
+    logger.info(f"Checking if {token} is in {secret_tokens}")
+
+    print(f"Headers in is_valid_management_token {headers}")
+    print(f"Headers in is_valid_management_token {headers}")
+
+    print(f"Checking if {token} is in {secret_tokens}")
```

Comment:
we probably don't want to log the token here, right?

Reply:
Correct, I ripped all of this out.